### DISCUSSION TITLE
How do I set a duration for a useMotionValue animation?

### DISCUSSION THREAD

**Discussion by GavanWilhite** [motiondivision/motion#2294] [Q&A]
I'm trying to animate a Chakra UI progress bar, but it uses a 'value' for the progress amount, which I think is conflicting with a Framer Motion property. So I'm doing this:

```
    const progress = useMotionValue(0);
    const progressValue = useTransform(progress, [0, 1], [0, 100], {
        ease: easeOut,
    });
    <Progress value={progressValue.get()} width='100%'></Progress>

```

It doesn't seem to be working though, and I can't figure out how I can set a duration. Thoughts?

**Comment by Albx68** [comment]
@GavanWilhite 
import { chakra } from "@chakra-ui/react"
import { motion, useMotionValue, useTransform, useSpring } from "framer-motion";
import { useEffect } from "react";


const ChakraMotionProgress = chakra(motion.div)

const ProgessIssue = () => {

    const progress = useMotionValue(0)
    const progressSpring = useSpring(progress, { stiffness: 40 })
    const animationDuration = 10000
    const width = 300

    useEffect(() => {
        let i
        i = setInterval(() => {
            progress.set(progress.get() + 10)
        }, 1000)
        const t = setTimeout(() => {
            clearInterval(i)
            clearTimeout(t)
        }, animationDuration)
        return () => clearInterval(i)
    }, [progress])

    const progressValue = useTransform(progressSpring, [0, 100], [0, width])

    return <>
        <motion.div
            style={{
                width: `${width}px`,
                margin: 'auto',
            }}
        >
            <ChakraMotionProgress
                style={{ width: progressValue }}
                h="20px"
                backgroundColor={"red"}
                borderRadius="md"
            />
            <motion.div>{progress}</motion.div>
        </motion.div>
    </>

}

export default ProgessIssue



You can try something like this, Here I've created an animated chakra progress bar using the chakra factory function, you can set a duration by playing around with the animationDuration variable  which is in ms

Let me know if you have any doubts or queries

Hope this helps :D

**Comment by GavanWilhite** [comment]
Thanks!
